src/Microsoft.Data.Analysis/DataFrameColumnCollection.cs

```diff
@@ -175,5 +175,278 @@ protected override void ClearItems()
             }
         }
 
+        /// <summary>
+        /// Searches for an <see cref="PrimitiveDataFrameColumn{T}"/> with the specified <paramref name="name"/> and attempts to return it as an <see cref="PrimitiveDataFrameColumn{T}"/>. If <typeparamref name="T"/> does not match <see cref="PrimitiveDataFrameColumn.DataType"/>, an exception is thrown.
```

Comment:
1. I don't know if I'd use the term `Searches` here. That kind of makes it sound like this is expensive, when it isn't.
2. I don't think we need to put `If <typeparamref name="T"/> does not match <see cref="PrimitiveDataFrameColumn.DataType"/>, an exception is thrown.` in the summary. Which exceptions are thrown and why should be in the `exception` section.

How about 

```
        /// <summary>
        /// Gets the <see cref="PrimitiveDataFrameColumn{T}"/> with the specified name.
        /// </summary>
        /// <param name="name">The name of the column.</param>
        /// <returns>The <see cref="PrimitiveDataFrameColumn{T}"/> with the specified name.</returns>
        /// <exception cref="ArgumentException">A column named <paramref name="name"/> cannot be found, or if the column's type doesn't match.</exception>
```

Reply:
1. Hmmm, I guess you are right. "Search" might lead to an impression that this is an expensive method. Maybe "Look up" would have sounded better and implied a O(1) op.

2. Cool. I'll change it.